count-things/javascript/stats.js

```diff
@@ -0,0 +1,25 @@
+const median = (items = [], order = (a, b) => b - a) => {
+  // https://www.tutorialspoint.com/calculating-median-of-an-array-javascript
+  const sorted = [...items].sort(order);
+  if (sorted.length % 2 === 0) {
+    const first = sorted[sorted.length / 2 - 1];
+    const second = sorted[sorted.length / 2];
+    return (first + second) / 2;
+  } else {
+    const mid = Math.floor(sorted.length / 2);
+    return sorted[mid];
+  }
+};
+
+const mean = (items = [], sum = (all, next) => all + next) =>
+  items.length === 0 ? 0 : items.reduce(sum, 0) / items.length;
+
+export const stats = (items = []) => {
+  const stats = {};
+
+  stats.median = median(items);
+  stats.mean = mean(items);
+  stats.stDeviation = Math.sqrt(mean(items.map((number) => (number - stats.mean) ** 2)));
```

Comment:
I'd personally be interested in a comment explaining why we want these three measurements (as in, what does we expect a median to "mean" here), but it's just curiosity.

Reply:
I don't know either. when in doubt, count?  It wasn't much more work to include it and says something about the code in a folder.

it could end up being interesting if your searching folders for an interesting one to sample, or if you're trying to draw some conclusions about how people write code differently in different directories

Reply:
ah okay, so we haven't actually tried computed those yet. I'm just wondering what the `stats.median` would actually be used for...I guess stuff like number of lines, number of variables, etc...

Reply:
right now I use this function to compute the average file & line lengths in a given folder.  mean, median, std, min & max are enough to get an idea of what's in each folder